EpiAware/docs/src/showcase/replications/chatzilena-2019/index.jl

```diff
@@ -0,0 +1,622 @@
+
+1. `deterministic_mdl`: This conditions the generative model on the data observation. We can sample from this model to find the posterior distribution of the parameters.
+2. `deterministic_uncond_mdl`: This _doesn't_ condition on the data. This is useful for prior and posterior predictive modelling.
+"
+
+# ╔═╡ dbc1b453-1c29-4f82-bec9-098d67f9e63f
+deterministic_mdl = deterministic_ode_mdl(data.in_bed, data.ts, obs, sir_prob, N);
+
+# ╔═╡ e795c2bf-0861-4e96-9921-db47f41af206
+deterministic_uncond_mdl = deterministic_ode_mdl(
+    fill(missing, length(data.in_bed)), data.ts, obs, sir_prob, N);
+
+# ╔═╡ e848434c-2543-43d1-ae22-5c4241f138bb
+md"
+We add a useful plotting utility.
+"
+
+# ╔═╡ ab8c98d1-d357-4c49-9f5a-f069e05c45f5
+function plot_predYt(data, gens; title::String, ylabel::String)
+    fig = Figure()
+    ga = fig[1, 1:2] = GridLayout()
+
+    ax = Axis(ga[1, 1];
+        title = title,
+        xticks = (data.ts[1:3:end], data.date[1:3:end] .|> string),
+        ylabel = ylabel
+    )
+    pred_Yt = mapreduce(hcat, gens) do gen
+        gen.obsYt
+    end |> X -> mapreduce(vcat, eachrow(X)) do row
+        quantile(row, [0.5, 0.025, 0.975, 0.1, 0.9, 0.25, 0.75])'
+    end
+
+    lines!(ax, data.ts, pred_Yt[:, 1]; linewidth = 3, color = :green, label = "Median")
+    band!(
+        ax, data.ts, pred_Yt[:, 2], pred_Yt[:, 3], color = (:green, 0.2), label = "95% CI")
+    band!(
+        ax, data.ts, pred_Yt[:, 4], pred_Yt[:, 5], color = (:green, 0.4), label = "80% CI")
+    band!(
+        ax, data.ts, pred_Yt[:, 6], pred_Yt[:, 7], color = (:green, 0.6), label = "50% CI")
+    scatter!(ax, data.in_bed, label = "data")
+    leg = Legend(ga[1, 2], ax; framevisible = false)
+    hidespines!(ax)
+
+    fig
+end
+
+# ╔═╡ 2c6ac235-e331-4189-8c8c-74de5f98b2c4
+md"
+**Prior predictive sampling**
+"
+
+# ╔═╡ a729f1cd-404c-4a33-a8f9-b2ea6f0adb62
+let
+    prior_chn = sample(deterministic_uncond_mdl, Prior(), 2000)
+    gens = generated_quantities(deterministic_uncond_mdl, prior_chn)
+    plot_predYt(data, gens;
+        title = "Prior predictive: deterministic model",
+        ylabel = "Number of Infected students"
+    )
+end
+
+# ╔═╡ 4c0759fb-76e9-4de5-9206-89e8bfb6c3bb
+md"
+The prior predictive checking suggests that _a priori_ our parameter beliefs are very far from the data. Approaching the inference naively can lead to poor fits.
+
+We do three things to mitigate this:
+
+1. We choose a switching ODE solver which switches between explicit (`Tsit5`) and implicit (`Rosenbrock23`) solvers. This helps avoid the ODE solver failing when the sampler tries extreme parameter values. This is the default `solver = AutoTsit5(Rosenbrock23())` above.
+2. To avoid the effect of numerically negative small values of `λt` we add a small `upjitter`.
+3. We locate the maximum likelihood point, that is we ignore the influence of the priors, as a useful starting point for `NUTS`.
+"
+
+# ╔═╡ 8d96db67-de3b-4704-9f54-f4ed50a4ecff
+nmle_tries = 100
```

Comment:
this makes me sad. Can't we do this using our EpiProblem framework? I think this might be out of scope for this PR and be a new issue as I can see it needing a feature?

Reply:
I think the answer is to write the infection model as a EpiAware infection model and then do directly?

Reply:
This would currently be a bit painful so maybe needs to wait for a refactor of the infection submodels

Reply:
That is a really good point:

1) https://github.com/CDCgov/Rt-without-renewal/issues/306 We have MAP/MLE opt as an issue
2) I could do this with the pathfinder functionality we already have

Reply:
As this works as is lets leave and update in an issue. I like the idea that we support multiple options. I think it makes sense though to also link out to the idea we have lots of options when we update this.